File: src/main/scala/com/github/dapeng/dms/dto/request/ListServiceRequest.scala
package com.github.dapeng.dms.dto.request

/**
  *
  * @author <a href=mailto:leihuazhe@gmail.com>maple</a>
  * @since 2018-11-09 10:54 AM
  */
case class ListServiceRequest(simpleName: Option[String],

                              serviceName: Option[String],

                              serviceId: Option[Long],

                              pageRequest: Option[PageRequest])


object Convert {

  import com.github.dapeng.dms.util.Implicits._

  def convertNull(request: ListServiceRequest): ListServiceRequest = {
    ListServiceRequest(request.simpleName.nullable,
      request.simpleName.nullable,
      request.serviceId.nullable,
      request.pageRequest.nullable)
  }

}

File: src/main/java/com/github/dapeng/dms/web/vo/request/ListServiceRequest.java
package com.github.dapeng.dms.web.vo.request;


/**
 * @author <a href=mailto:leihuazhe@gmail.com>maple</a>
 * @since 2018-11-09 1:06 PM
 */
public class ListServiceRequest {

    private String simpleName;

    private String serviceName;

    private Long serviceId;

    private DmsPageRequest pageRequest;

    public String getSimpleName() {
        return simpleName;
    }

    public String getServiceName() {
        return serviceName;
    }

    public Long getServiceId() {
        return serviceId;
    }

    public DmsPageRequest getPageRequest() {
        return pageRequest;
    }
}
